upstairs/src/lib.rs

```diff
@@ -4546,21 +4590,71 @@ impl Upstairs {
         let next_id = downstairs.next_id();
 
         /*
-         * Now create a downstairs work job for each (eid, bo, len) returned
-         * from extent_from_offset
+         * Walk the downstairs work active list, and pull out all the active
+         * jobs (anything we have not submitted back to the guest). Construct
+         * a list of dependencies for this read based on the
+         * following rules:
+         *
+         * - reads do not depend on flushes, only writes (because flushes do
+         *   not modify data!)
+         * - any overlap of impacted blocks requires a dependency
+         *
+         * TODO: scanning backwards, any overlap of impacted blocks will
+         * create a dependency. take this an example:
+         *
+         *       block
+         * op# | 0 1 2 | deps
+         * ----|-------------
+         *   0 | R     |
+         *   1 | R     | 0
+         *   2 | R     | 0,1
```

Comment:
This example is not correct, is it?  I thought reads don't depend on overlapping reads, only on overlapping writes?

Reply:
you're right, removed in 4691ec5